components/BlogListing.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { ISbRichtext, SbBlokData, storyblokEditable } from '@storyblok/react';
+import BlogPost, { SbBlogPostProps } from './BlogPost';
+import { Author } from '../types';
+import { getStoryblokApi } from '@storyblok/react';
+import Link from 'next/link';
+import Pagination from "./Pagination"
+
+interface BlogPostData extends SbBlokData, SbBlogPostProps {
+  title: string;
+  content: ISbRichtext;
+  author: Author;
+  media: any;
+  description: string;
+}
+
+interface BlogListingProps {
+  // Add any props you need here
+}
+
+const BlogListing: React.FC<BlogListingProps> = () => {
+  const [posts, setPosts] = useState<BlogPostData[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const postsPerPage = 6;
+
+  useEffect(() => {
+    const fetchAllPosts = async () => {
+      try {
+        const storyblokApi = getStoryblokApi();
+        const { data } = await storyblokApi.getStories({
+          content_type: 'blogPost',
+        });
+        const allPosts: BlogPostData[] = data.stories.map((post: any) => ({
+          title: post.content.title,
+          content: post.slug,
+          author: post.content.author,
+          media: post.content.media.filename,
+          description: post.content.description,
+        }));
+        setPosts(allPosts);
+      } catch (error) {
+        console.error('Error fetching blog posts:', error);
+      }
+    };
+
+    fetchAllPosts();
+  }, []);
+
+  const totalPages = Math.ceil(posts.length / postsPerPage);
+  const indexOfLastPost = currentPage * postsPerPage;
+  const indexOfFirstPost = indexOfLastPost - postsPerPage;
+  const currentPosts = posts.slice(indexOfFirstPost, indexOfLastPost);
```

Comment:
@Nancy-Kaur11 Thanks for adding pagination. However, Pagination should be done via `getStories` function.
getStories supports pagination and items per page.

Fetching all posts and calculating on client side is not recommended on production projects and pagination should be always done via API.

Every storyblok API call supports pagination 
https://www.storyblok.com/docs/api/content-delivery/v2/getting-started/pagination